lib/api/checkr_api.py

```diff
@@ -0,0 +1,128 @@
+import json
+from datetime import date
+from typing import Any, Optional
+
+import requests
+from requests import Session
+from requests.adapters import HTTPAdapter, Retry
+from requests.auth import HTTPBasicAuth
+
+from lib.helpers.utils import get_default_retries
+
+
+class CheckrAPI:
+    def __init__(
+        self,
+        secret_api_key: str,
+        api_url: str,
+        session: Session = Session(),
+        retries: Retry = get_default_retries(),
```

Comment:
Because python is pass-by-value-reference these optional parameters should be as follows:
```suggestion
        session: Optional[Session] = None,
        retries: Optional[Retry] = None,
```

Then you set them in the `__init__` body like so:

```python
self.session = session or Session()
self.retries = retries or get_default_retries()
```

This stops the same object from being set (since the invocation of `get_default_retries()` is fired _at class definition_ and thus, is shared across instances in your solution, so it could bleed state between instances. 

In general, only use default parameters that are non-collection primitives like `int` `str` `bool` `float`

Reply:
Thank you for the explanation! I'll change it.